packages/roosterjs-editor-plugins/lib/plugins/HyperLink/HyperLink.ts

```diff
@@ -67,6 +81,43 @@ export default class HyperLink implements EditorPlugin {
      * @param event PluginEvent object
      */
     public onPluginEvent(event: PluginEvent): void {
+        if (
+            event.eventType == PluginEventType.MouseUp ||
+            (event.eventType == PluginEventType.KeyUp && !this.isContentEditValue(event.rawEvent))
```

Comment:
I think you also need to handle ContentChanged event

Reply:
Thanks, I tested that the behavior is triggered when pasting content at the end of an edited link

Reply:
ContentChanged event can be triggered not only from paste, but also a lot of reasons, even from other plugins.

Reply:
That's fair. Should we be selective on when it should trigger? I think it would make sense to trigger the behavior for Paste, Cut, and Drop (and maybe SetContent? I didn't see any weird behavior with undo logic when testing).

Reply:
I think it should be fine to handle everything.